File: matlab/common/Holding.m
%% header
% handle a holding and the related logic
% date: 7/3/2015

%%
% it is a handle class
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% not needed 
% classdef Holding < handle
%     properties
%         StockID         % the stockID held
%         Weight          % weight in a portfolio, could be negative
%         Shares          % number of shares held, could be negative or null
%         MarketValue     % the market value of the holding, could be negative or null
%     end
%     
%     methods
%         function obj = Holding(n)
%             
%         end
%         
%         function obj = Holding(stock, weight, shares, marketValue)
%             obj.Stock = stock;
%             obj.Weight = weight;
%             if(nargin>2) 
%                 obj.Shares = shares;
%                 obj.MarketValue = marketValue;
%             end
%         end        
%         
%     end
%     
% end

File: common/Holding.py
# %% header
# % handle a holding and the related logic
# % date: 7/3/2015

# %%
# % it is a handle class
# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% not needed


class Holding():
    def __init__(self, stock, weight, shares=None, marketValue=None):
        self.StockID = stock              # the stockID held
        self.Weight = weight              # number of shares held, could be negative or null
        self.Shares = shares
        self.MarketValue = marketValue    #the market value of the holding, could be negative or null

